lib/api.ts

```diff
@@ -658,6 +658,15 @@ module.exports = function (PersistObjectTemplate, baseClassForPersist) {
             }
         };
 
+        /**
+         * Can generate object id even before saving the record to the database.
+         * @returns {string}
+         */
+        template.prototype.generateId =  function (): string {
+            var persistObjectTemplate = this.__objectTemplate__ || self;
```

Comment:
nit: can we start going forward using let/const?

Reply:
@hackerrdave - Agree that we should go with let, but when I see all the remaining vars, I tend to  ignore that.. anyways, @sshankar19 and @nsaenz7 are working on restructuring the code, I think they are going to cover..

Reply:
Done